dotnet/samples/Demos/Hosting/Agent/ChatWithAgent.Web/AgentCompletionsApiClient.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+namespace ChatWithAgent.Web;
+
+internal sealed class AgentCompletionsApiClient(HttpClient httpClient)
+{
+    public async Task<string> CompleteAsync(string prompt, CancellationToken cancellationToken)
+    {
+        var result = await httpClient.PostAsJsonAsync<AgentCompletionRequest>("/agent/completions", new AgentCompletionRequest() { Prompt = prompt }, cancellationToken).ConfigureAwait(false);
+
+        result.EnsureSuccessStatusCode();
+
+        return await result.Content.ReadAsStringAsync(cancellationToken).ConfigureAwait(false);
+    }
+}
+
+/// <summary>
+/// The agent completion request model.
+/// </summary>
+internal sealed class AgentCompletionRequest
```

Comment:
Split in another file or make it a private sealed class.

```suggestion
internal sealed class AgentCompletionRequest
```